src/main/java/com/questhelper/quests/tribaltotem/TribalTotem.java

```diff
@@ -0,0 +1,164 @@
+import com.questhelper.steps.QuestStep;
+
+@QuestDescriptor(
+        quest = QuestHelperQuest.TRIBAL_TOTEM
+)
+public class TribalTotem extends BasicQuestHelper
+{
+    ItemRequirement coins, amuletOfGlory, ardougneTeleports, addressLabel, totem;
+
+    QuestStep talkToKangaiMau, investigateCrate, useLabel, talkToEmployee, talkToCromperty, enterPassword, solvePassword, climbStairs, searchChest, leaveHouse, talkToKangaiMauAgain;
+
+    ConditionForStep hasLabel, inEntrance, inMiddleRoom, openedLockWidget, inStairway, investigatedStairs, isUpstairs, chestOpened, hasTotem;
+
+    Zone houseGroundFloorEntrance, houseGroundFloorMiddleRoom, houseGroundFloor, houseFirstFloor;
+
+    @Override
+    public Map<Integer, QuestStep> loadSteps()
+    {
+        loadZones();
+        setupItemRequirements();
+        setupConditions();
+        setupSteps();
+
+        ConditionalStep useLabelOnCrate = new ConditionalStep(this, investigateCrate);
+        useLabelOnCrate.addStep(hasLabel, useLabel);
+
+        ConditionalStep navigateMansion = new ConditionalStep(this, talkToCromperty);
+        navigateMansion.addStep(hasTotem, talkToKangaiMauAgain);
+        navigateMansion.addStep(new Conditions(openedLockWidget, inMiddleRoom), solvePassword);
+        navigateMansion.addStep(inStairway, climbStairs);
+        navigateMansion.addStep(isUpstairs, searchChest);
+        navigateMansion.addStep(inEntrance, enterPassword);
+        navigateMansion.addStep(inMiddleRoom, enterPassword);
+
+        return new ImmutableMap.Builder<Integer, QuestStep>()
+                .put(0, talkToKangaiMau)
+                .put(1, useLabelOnCrate)
+                .put(2, talkToEmployee)
+                .put(4, navigateMansion)
+                .build();
+    }
+
+    public void setupItemRequirements()
+    {
+        coins = new ItemRequirement("Coins or more for boat trips", ItemID.COINS, 90);
+        amuletOfGlory = new ItemRequirement("Amulet of glory", ItemCollections.getAmuletOfGlories());
+        ardougneTeleports = new ItemRequirement("Ardougne teleports", ItemID.ARDOUGNE_TELEPORT);
+        addressLabel = new ItemRequirement("Address label", ItemID.ADDRESS_LABEL);
+        totem = new ItemRequirement("Totem", ItemID.TOTEM);
+    }
+
+    @Override
+    public ArrayList<ItemRequirement> getItemRecommended()
+    {
+        ArrayList<ItemRequirement> reqs = new ArrayList<>();
+        reqs.add(coins);
+        reqs.add(amuletOfGlory);
+        reqs.add(ardougneTeleports);
+        return reqs;
+    }
+
+    public void loadZones()
+    {
+        houseGroundFloorEntrance = new Zone(new WorldPoint(2637, 3320, 0), new WorldPoint(2639, 3325, 0));
+        houseGroundFloorMiddleRoom = new Zone(new WorldPoint(2634, 3322, 0), new WorldPoint(2636, 3324, 0));
+        houseGroundFloor = new Zone(new WorldPoint(2629, 3321, 0), new WorldPoint(2633, 3325, 0));
+        houseFirstFloor = new Zone(new WorldPoint(2630, 3319, 1), new WorldPoint(2638, 3323, 1));
```

Comment:
The first floor needs to extend a tile further south and east to capture the entire first floor.